sequence-diagrams/host-provisioning-xPU-VF.md

```diff
@@ -47,54 +48,73 @@ as a Kubernetes deployment.
+### Multi Cluster sequence diagrams
 
-TODO
+#### Initialization
 
-#### Advantages
+The following high level diagram provides an overview of the initialization of the entities involved in provisioning an xPU VF to a host.
 
-The main advantage of this approach is that the scope of what is shared across clusters can be limited (in the two cluster scenario), while also allowing us to maintain a relatively simple CNI on the host side.
+![host-provisioning-initialization](./images/host-provisioning-xPU-VF-init.png)
 
-#### Required CRDs
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-TODO (WIP)
+![host-provisioning-initialization-no-host-agent](./images/host-provisioning-xPU-VF-init-no-host-agent.png)
 
-- Netdev (New)
-- Service: reflecting the services in the tenant cluster.
-- EndpointSlice: reflecting the endpoint slices in the tenant cluster.
-- Network policy
+#### Pod Creation
 
-### Single Cluster
+![host-provisioning-pod-creation](./images/host-provisioning-xPU-VF-pod-creation.png)
 
-The following sequence diagram provides an overview of the entities working
-together to provision an interface for a pod in the single cluster scenario.
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-![host-provisioning-xPU-VF](./images/host-provisioning-xPU-VF.png)
+![host-provisioning-pod-creation-no-host-agent](./images/host-provisioning-xPU-VF-pod-creation-no-host-agent.png)
 
-### Multi Cluster
+#### Pod Deletion
 
-The following sequence diagram provides an overview of the entities working
-together to provision an interface for a pod in the Multi cluster scenario.
+![host-provisioning-pod-deletion](./images/host-provisioning-xPU-VF-pod-deletion.png)
 
-![host-provisioning-xPU-VF-multi-cluster](./images/host-provisioning-xPU-VF-multi-cluster.png)
+An alternative option would be to eliminate the host agent, and the sequence diagram for that looks as follows:
 
-## CRDs
+![host-provisioning-pod-deletion-no-host-agent](./images/host-provisioning-xPU-VF-pod-deletion-no-host-agent.png)
 
-## DPUNetworkNodeState
+## CRDs
 
-```yaml
-TODO WIP
+### Netdev
+
+```golang
+type NetdevStatus string
+
+const (
+    Connected NetdevStatus = "connected"
+    Connecting NetdevStatus = "connecting"
+    Disconnected NetdevStatus = "disconnected"
+    ConnectionError NetdevStatus = "error"
+)
+
+type NetdevSpec struct {
+    NetdevID string `json:"netdev_id"` // SerialNum_PFID_SFID
+    Hostname string `json:"hostname"`
+    Subnets []string `json:"subnets"` // or maybe other routing info?
+    IP string `json:"ip"`
+    Backend string `json:"backend,omitempty"` //subfunction or virtual function
+    Status NetdevStatus `json:"status"`
+}
+
+type Netdev struct {
+    metav1.TypeMeta `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    Spec NetdevSpec `json:"spec"`
```

Comment:
Probably want status as a separate structure, that way Host owns Spec and DPU owns Status. Question is whether IP belongs in Spec or Status (depending on whether Host or DPU writes it).
```suggestion
    Spec NetdevSpec `json:"spec"`
    // +optional
    Status NetdevStatus `json:"status,omitempty"`
```

Reply:
makes sense